.github/scripts/explanations_merlin_down

```diff
@@ -0,0 +1,16 @@
+function definition	hdb_catalog	hdb_catalog.hdb_catalog."notify_hasura_refreshResourceTypes_INSERT"()	missing in 1st DB	From_RefreshResourceTypes_Action	2
```

Comment:
Not sure I quite understand what's going on here - I remember that pgcmp uses "explanations" to dismiss certain expected differences - and that we do have expected differences, since we're comparing a hasura-less environment to one with hasura running.

What I don't quite follow is what this new file's role is - and how it is or isn't different from refreshActivityTypes. Could you catch me up here?

Reply:
Sure!

Contents of the file: 
In order to support the new Hasura Event (`refreshActivityTypes`), Hasura automatically creates new functions/triggers in its schema (`hdb_catalog`) on the relevant table in the relevant database (`mission_model` in `aerie_merlin`). I believe these objects should go away if we added checking out the v1.0.1 and restarting Hasura to update the metadata to the `dump-current` job. However, that would add at minimum 30 seconds to the job's run time, which is already ~3 mins. 

Reason why this file exists: 
When providing an explanations file, [pgcmp will consider an explanation a "bad explanation" if the explanation doesn't match the state of the DB](https://github.com/cbbrowne/pgcmp#pgcmp-1) (ie, if you say something is missing in the first DB, but it's actually present in both). The original `explanations.txt` were things that needed to be explained in all DBs for all the "down" comparisons (note [that `explanations.txt` isn't used in the "up" comparisons](https://github.com/NASA-AMMOS/aerie/blob/81deeb9fdf1c043178dd24ee4c7ca99c6b7509c4/.github/workflows/pgcmp.yml#L278-L283), since both DBs should have a `migrations` schema). These new explanations are only needed when comparing the Merlin Base and Merlin Migrated to Base DBs, so they had to be put in their own file. To avoid duplication, the file only contains the unique explanations, and the workflow concatenates them together to get the complete list of explanations for the Merlin DB in the `pgcmp-down` job.

For additional clarity:
> since we're comparing a hasura-less environment to one with hasura running

Both of the DBs had Hasura running. However, the metadata changes in this PR required Hasura to add Postgres objects (functions and triggers on the `mission model` table), which PGCMP notices. (In general, our metadata goes into a table the `aerie_hasura` database, which both isn't compared and maintains the same structure, pg-objects-wise)

> and how it is or isn't different from refreshActivityTypes

Hopefully the above explanation made it clear, but just in case: this isn't different from `refreshActivityTypes`, they're the Postgres Objects needed for it to work.

Reply:
Thanks for the explanation :+1: I appreciate you framing the tradeoff between simplicity and the time it takes to run this test